### DISCUSSION TITLE
fixed positioned <DropdownMenu.Content>

### DISCUSSION THREAD

**Discussion by fnick851** [radix-ui/primitives#971] [Help]
I notice <DropdownMenu.Content> has absolute position with left: 0 and top: 0, but it is always placed as child of <body> and appears under the <DropdownMenu.Trigger>. How is this achieved? It is not a child of <DropdownMenu.Trigger>, but it is always absolute positioned relative to the Trigger?

Also, is there a way to make it fixed positioned? I need to build a floating navigation on the side of the page. Thanks.

**Comment by benoitgrelard** [comment]
Hi @fnick851,

> it is always placed as child of and appears under the <DropdownMenu.Trigger>. How is this achieved? 

I assume when you say this, you mean in terms of the way you render the parts together?
The way it works is that the content is rendered in a portal elsewhere in the DOM (not right after the trigger) and it is then kept positioned relatively to the trigger. This is so we can handle things like layering as well as collisions.

> Also, is there a way to make it fixed positioned? I need to build a floating navigation on the side of the page. Thanks.

Can you provide more details on what you're trying to achieve here?
The whole point of a dropdown menu is to have the menu attached to a trigger, so I'm struggling to understand why you'd want it positioned fixed.

**Reply by fnick851** [reply]
@benoitgrelard My apologies the explanation is a bit confusing. 

So I'm just looking at the very basic example on the documentation page. And I did notice it is rendered via a Portal. That's why I don't understand how its position is always relative to the trigger, because with a Portal, it renders as the absolute positioned child of the \<body>, not the child of the trigger? Isn't absolute positioned element positions relative to the first "relative" positioned parent element it finds?

**Reply by benoitgrelard** [reply]
We have logic that positions it absolutely but calculates its position based on where the trigger is, and based on the space available on the side you requested it to open, to avoid things like collisions, etc. Does that make sense?

**Reply by fnick851** [reply]
@benoitgrelard That makes sense now. Thanks. 

So my scenario is I need a "sticky" navigation so that when I scroll the page, both the trigger and the dropdown content remains fixed or sticky positioned (won't scroll away with the page). Is there a way to achieve this? I can set position attribute of the Root (&Trigger). But it won't affect the Content.

**Reply by benoitgrelard** [reply]
Ah gotcha, ideally that should work straight off the bat, because the trigger isn't moving.
However, we currently have an issue with this specifically when the trigger is fixed.

Here's an open issue about it: https://github.com/radix-ui/primitives/issues/781

This is with our `Popover` but they both share the same underlying mechanics for positioning so it's relevant for both.
Here (https://github.com/radix-ui/primitives/issues/781#issuecomment-915255150) someone is trying to do essentially what you want.
And here (https://github.com/radix-ui/primitives/issues/781#issuecomment-915976878) is a potential workaround for now.

**Reply by fnick851** [reply]
Got it. Thanks for the information!
